go/host/host.go

```diff
@@ -659,49 +659,58 @@ func (a *Node) processSharedSecretResponse(_ *ethadapter.L1RespondSecretTx) erro
 
 // monitors the L1 client for new blocks and injects them into the aggregator
 func (a *Node) monitorBlocks() {
-	listener := a.ethClient.BlockListener()
+	listener, subs := a.ethClient.BlockListener()
 	common.LogWithID(a.shortID, "Start monitoring Ethereum blocks..")
 
 	// only process blocks if the node is running
 	for atomic.LoadInt32(a.stopNodeInterrupt) == 0 {
-		blkHeader := <-listener
+		select {
+		case err := <-subs.Err():
+			log.Error("Restarting L1 block monitoring. Errored with: %s", err)
+			listener, subs = a.ethClient.BlockListener()
```

Comment:
I would add a todo here, as there might be missing blocks which need to be fetched to get back up to date

Reply:
Good point, this is a very simplistic happy view of catching up.
Added 👍